### PR TITLE
tests: add support for before/after scripts to the DAG verifier.

### PR DESCRIPTION
**Pull request by isidentical**

### CODE REVIEW

File: integration_tests/features/steps/fal_steps.py [hunk 1]
```diff
@@ -252,7 +257,8 @@ def _verify_node_order(context):
     assert_precedes = partial(_assert_precedes, ordered_nodes)
     assert_succeeds = partial(_assert_succeeds, ordered_nodes)
     for node in ordered_nodes:
-        if _is_script(node):
```
Comment (chamini2) [inline integration_tests/features/steps/fal_steps.py:255]: you can delete this function now

File: integration_tests/features/steps/fal_steps.py [hunk 2]
```diff
@@ -252,7 +257,8 @@ def _verify_node_order(context):
...
+        # Skip all the nodes that are not part of the graph.
```
Comment (chamini2) [inline integration_tests/features/steps/fal_steps.py:260]: this happens?

Comment (isidentical) [reply integration_tests/features/steps/fal_steps.py:260]: `ordered_nodes` contain both 'real scripts' and `pre-hooks`/`post-hooks` which are not part of the graph.

File: integration_tests/features/steps/fal_steps.py (hunk 1 above)
Comment (isidentical) [reply integration_tests/features/steps/fal_steps.py:240]: done!

**Review by chamini2 — APPROVED** [review]

**Pull request merged**